### PR TITLE
Fix issue where users can't logon if there is a space before of behin…

### LINKED ISSUE #119: Trim username on login
When logging in, the username should be trimmed

### PR DESCRIPTION
**Pull request by pilaar**
…d the username: resolves #119

### CODE REVIEW

File: src/components/authentication.js [hunk 1]
```diff
@@ -71,7 +71,7 @@ export class Authentication {
         return this.router.navigate('login');
     };
 
-    async login(username, password, extraParameters, storeCredentials=false) {
+    async login(username.trim(), password, extraParameters, storeCredentials=false) {
```
Comment (khenderick) [inline src/components/authentication.js]: Does that work when you call `.trim()` in the function arguments?

```
let foo = async (x.trim()) => { console.log(`x: -${x}-`); }
VM128:1 Uncaught SyntaxError: Unexpected token .
```

Comment (pilaar) [reply src/components/authentication.js]: Yes that works

Comment (khenderick) [reply src/components/authentication.js]: When I start a build I can't get it to run:

```
ERROR in ./src/components/authentication.js
Module build failed: SyntaxError: Unexpected token, expected , (74:24)

  72 |     };
  73 | 
> 74 |     async login(username.trim(), password, extraParameters, storeCredentials=false) {
     |                         ^
  75 |         let data = await this.api.login(username, password, extraParameters, {ignore401: true});
  76 |         if (data['next_step'] !== undefined) {
  77 |             return data;
```

File: src/components/authentication.js [hunk 2]
```diff
@@ -72,7 +72,7 @@ export class Authentication {
     };
 
     async login(username, password, extraParameters, storeCredentials=false) {
-        let data = await this.api.login(username, password, extraParameters, {ignore401: true});
+        let data = await this.api.login(username.trim(), password, extraParameters, {ignore401: true});
```
Comment (khenderick) [inline src/components/authentication.js]: Would make sense to do the same with the password, as it seems that when creating a user, the password is also trimmed. Also, these variables are used below to put in a `PasswordCredential` when available. Wouldn't it make more sense to store the trimmed values over there?

File: src/components/authentication.js [hunk 3]
```diff
@@ -72,6 +72,8 @@ export class Authentication {
     };
 
     async login(username, password, extraParameters, storeCredentials=false) {
+        username = username.trim()
```
Comment (khenderick) [inline src/components/authentication.js]: You forgot a `;`.

**Pull request merged**